torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1,165 +1,73 @@
-        PIL.Image.Image: _F.affine,
-        features.Image: K.affine_image,
-    }
-)
-def affine(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.rotate,
-        PIL.Image.Image: _F.rotate,
-        features.Image: K.rotate_image,
-    }
-)
-def rotate(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.pad,
-        PIL.Image.Image: _F.pad,
-        features.Image: K.pad_image,
-    }
-)
-def pad(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.crop,
-        PIL.Image.Image: _F.crop,
-        features.Image: K.crop_image,
-    }
-)
-def crop(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.perspective,
-        PIL.Image.Image: _F.perspective,
-        features.Image: K.perspective_image,
-    }
-)
-def perspective(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.vflip,
-        PIL.Image.Image: _F.vflip,
-        features.Image: K.vertical_flip_image,
-    }
-)
-def vertical_flip(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.five_crop,
-        PIL.Image.Image: _F.five_crop,
-        features.Image: K.five_crop_image,
-    }
-)
-def five_crop(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
-
-
-@dispatch(
-    {
-        torch.Tensor: _F.ten_crop,
-        PIL.Image.Image: _F.ten_crop,
-        features.Image: K.ten_crop_image,
-    }
-)
-def ten_crop(input: Any, *args: Any, **kwargs: Any) -> Any:
-    """TODO: add docstring"""
-    ...
+from torchvision.transforms import functional as _F, InterpolationMode
+
+from ._meta_conversion import convert_bounding_box_format
+
+
+horizontal_flip_image = _F.hflip
+
+
+def horizontal_flip_bounding_box(
+    bounding_box: torch.Tensor, *, format: features.BoundingBoxFormat, image_size: Tuple[int, int]
+) -> torch.Tensor:
+    shape = bounding_box.shape
+
+    bounding_box = convert_bounding_box_format(
+        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY
+    ).view(-1, 4)
+
+    bounding_box[:, [0, 2]] = image_size[1] - bounding_box[:, [2, 0]]
+
+    return convert_bounding_box_format(
+        bounding_box, old_format=features.BoundingBoxFormat.XYXY, new_format=format
+    ).view(shape)
+
+
+def resize_image(
+    image: torch.Tensor,
+    size: List[int],
+    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
+    max_size: Optional[int] = None,
+    antialias: Optional[bool] = None,
+) -> torch.Tensor:
+    new_height, new_width = size
+    num_channels, old_height, old_width = image.shape[-3:]
```

Comment:
Above the `horizontal_flip_image` kernel uses the old `_F.hflip` which handles both Images and PIL. Nevertheless, here the `resize_image` doesn't handle PIL. This is somehow inconsistent. We shouldn't have half of the methods supporting PIL and half not doing so.

Moreover I think we need to phase out the idiom `num_channels, old_height, old_width = image.shape[-3:]` from the code in favour of calling the correct kernels such as `get_image_size` and `get_image_num_channels`.

Reply:
Reminder to change the image size calculations to always call the above kernels.

Reply:
@pmeier I was wrong here.

The new get_image_size/get_image_num_channels that live in utilities are not jit-scriptable (due to the fact they know about PIL and `Image`). So here you will either use the `_FT` versions (which is what you did) or do `num_channels, old_height, old_width = image.shape[-3:]` as you proposed originally.

I think the latter makes more sense once and I'll restore it on a follow up PR.